code/AssetLib/glTF2/glTF2Exporter.cpp

```diff
@@ -836,7 +836,7 @@ void glTF2Exporter::ExportMaterials() {
             GetMatTex(mat, m->pbrMetallicRoughness.baseColorTexture, aiTextureType_DIFFUSE);
         }
 
-        GetMatTex(mat, m->pbrMetallicRoughness.metallicRoughnessTexture, AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_METALLICROUGHNESS_TEXTURE);
+        GetMatTex(mat, m->pbrMetallicRoughness.metallicRoughnessTexture, aiTextureType_DIFFUSE_ROUGHNESS);
```

Comment:
In the importer `AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_METALLICROUGHNESS_TEXTURE`, `aiTextureType_METALNESS`, and `aiTextureType_DIFFUSE_ROUGHNESS` are all set to `pbrMetallicRoughness.metallicRoughnessTexture`. Would it make sense to also try all three here for backwards compatibility, having a similar fallback strategy as above for the `pbrMetallicRoughness.baseColorTexture`?

Reply:
It looks like what you said makes sense, and I have followed your suggestion. Additionally, I took the opportunity to format the code.

Reply:
Thanks for the adjustments. Using `!= AI_SUCCESS` checks may be more consistent with the rest of the file though. Can you please also verify, whether it is better to first try `aiTextureType_METALNESS` or `aiTextureType_DIFFUSE_ROUGHNESS`?

Reply:
I'm sorry, but the GetMatTex function seems to not have a return value, and I have not found any interface with an aiReturn return value that has the same functionality as GetMatTex.
About the order: I have tried exporting according to aiTextureType_METALNESS, aiTextureType_DIFFUSE_ROUGHNESS, and exporting with both together. In the exported glb file, there is only 'pbrMetallicRoughness', which should be consistent with the GLB file specification. I also attempted to re-import with Assimp, and regardless of whether I exported with aiTextureType_METALNESS, aiTextureType_DIFFUSE_ROUGHNESS, or both, I was always able to read the output using either aiTextureType_DIFFUSE_ROUGHNESS or aiTextureType_METALNESS. I further tested by importing the models exported with these methods into Blender to check. In Blender, the material effects of each model were consistent. Thus, I believe the order of aiTextureType_METALNESS and aiTextureType_DIFFUSE_ROUGHNESS should not be crucial.

Reply:
Okay, thanks a lot for digging in!